File: frontend/src/app/core/note.ts
export interface Note {
  id: number;
  title: string;
  content: string;
  color: string | null;
  is_pinned: boolean;
  is_deleted: boolean;
  created_at: string;
  updated_at: string;
  summary: string | null;
}

export interface NoteCreate {
  title: string;
  content: string;
  color?: string | null;
  is_pinned?: boolean;
}

export interface NoteUpdate {
  title?: string;
  content?: string;
  color?: string | null;
  is_pinned?: boolean;
}

File: backend/src/schemas/note.py
from pydantic import BaseModel
from datetime import datetime

class NoteBase(BaseModel):
    title: str
    content: str
    color: str | None = None

class NoteCreate(NoteBase):
    is_pinned: bool = False

class NoteUpdate(BaseModel):
    title: str | None = None
    content: str | None = None
    color: str | None = None
    is_pinned: bool | None = None
    is_deleted: bool | None = None

class NoteOut(NoteBase):
    id : int
    is_pinned: bool = False
    is_deleted: bool = False
    created_at: datetime
    updated_at: datetime
    summary: str | None = None